paasta_tools/docker_wrapper.py

```diff
@@ -71,31 +76,64 @@ def generate_hostname(fqdn, mesos_task_id):
     return hostname
 
 
-def add_hostname(args, hostname):
-    # Add --hostname argument immediately after 'run' command if it exists
+def add_argument(args, argument):
+    # Add an argument immediately after 'run' command if it exists
     args = list(args)
 
     try:
         run_index = args.index('run')
     except ValueError:
         pass
     else:
-        args.insert(run_index + 1, '--hostname=' + hostname)
+        args.insert(run_index + 1, argument)
 
     return args
 
 
+def get_core_list(cpuid):
+    core = 0
+    core_list = []
+
+    try:
+        with open('/proc/cpuinfo', 'r') as f:
+            for line in f:
+                m = re.match('physical\sid.*(\d)', line)
+                if m:
+                    if int(m.group(1)) == cpuid:
+                        core_list.append(core)
+                    core += 1
+    except IOError:
+        pass
+
+    return core_list
+
+
+def is_numa_enabled():
+    return os.path.exists('/proc/1/numa_maps')
+
+
 def main(argv=None):
     argv = argv if argv is not None else sys.argv[1:]
 
     env_args = parse_env_args(argv)
-    fqdn = socket.getfqdn()
 
     # Marathon sets MESOS_TASK_ID whereas Chronos sets mesos_task_id
     mesos_task_id = env_args.get('MESOS_TASK_ID') or env_args.get('mesos_task_id')
 
+    # Invalid the variable if it has a bogus value
+    try:
+        numa_cpuid = int(env_args.get('NUMA_CPU_AFFINITY'))
```

Comment:
This still needs some thought on *how* this will actually work in real life when mesos doesn't know that it is scheduling 2 tasks on the same host that have `NUMA_CPU_AFFINITY=0`.